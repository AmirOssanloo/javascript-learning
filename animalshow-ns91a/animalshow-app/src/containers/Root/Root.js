import React, { useState } from 'react';
import styled from 'styled-components';
import AnimalDisplay from '#components/AnimalDisplay';
import AnimalQuery from '#components/AnimalQuery';

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: inherit;
  height: inherit;
`;

const Root = () => {
  const [showResult, setShowResult] = useState(false);

  return (
    <Wrapper>
      {showResult
        ? <AnimalDisplay onChangeToAnimalQuery={() => setShowResult(false)} />
        : <AnimalQuery onChangeToAnimalDisplay={() => setShowResult(true)} />}
    </Wrapper>
  );
};

export default Root;
