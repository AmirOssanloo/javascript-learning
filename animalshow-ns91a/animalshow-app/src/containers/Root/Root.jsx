import React, { useState } from 'react';
import styled from 'styled-components';
import AnimalDisplay from '#components/AnimalDisplay';
import AnimalQuery from '#components/AnimalQuery';

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  width: inherit;
  height: inherit;
`;

const Content = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: 40rem;
  padding-bottom: 10rem;
`;

const Title = styled.h1`
  color: ${({ theme }) => theme.primary};
  font-size: 4.8rem;
`;

const Root = () => {
  const [showResult, setShowResult] = useState(false);

  return (
    <Wrapper>
      <Content>
        <Title>Animal Query</Title>
        {showResult
          ? <AnimalDisplay onChangeToAnimalQuery={() => setShowResult(false)} />
          : <AnimalQuery onChangeToAnimalDisplay={() => setShowResult(true)} />}
      </Content>
    </Wrapper>
  );
};

export default Root;


