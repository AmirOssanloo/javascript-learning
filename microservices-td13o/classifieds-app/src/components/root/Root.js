import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import gql from 'graphql-tag';
import { useDispatch } from 'react-redux';
import AccountDetails from '#root/components/Root/AccountDetails';
import Listings from '#root/components/Root/Listings';
import graphqlClient from '#root/api/graphqlClient';
import { setSession } from '#root/store/ducks/session';

const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  padding: 1rem;
  box-sizing: border-box;
`;

const Container = styled.div`
  display: flex;
  flex-flow: row nowrap;
  width: 40rem;
  margin: 0 auto;
`;

const Content = styled.div`
  flex: 1;
  margin-right: 1rem;
`;

const Sidebar = styled.div`
  flex: 0 auto;
  width: 10rem;
`;

const query = gql`
  {
    userSession(me: true) {
      id
      user {
        id
        email
      }
    }
  }
`;

const Root = () => {
  const dispatch = useDispatch();
  const [initialised, setInitialised] = useState(false);

  useEffect(() => {
    graphqlClient.query({ query }).then(({ data }) => {
      if (data.userSession) {
        dispatch(setSession(data.userSession));
      }

      setInitialised(true);
    })
  }, []);

  if (!initialised) {
    return <h1>Loading...</h1>
  }

  return (
    <Wrapper>
      <Container>
        <Content>
          <Listings />
        </Content>
        <Sidebar>
          <AccountDetails />
        </Sidebar>
      </Container>
    </Wrapper>
  );
};

export default Root;
