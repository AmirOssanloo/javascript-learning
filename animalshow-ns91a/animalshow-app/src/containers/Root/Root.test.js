import React from 'react';
import { shallow } from 'enzyme';
import Root from './Root';

describe('Root', () => {
  test('should match with snapshot', () => {
    const component = shallow(<Root />);
    expect(component).toMatchSnapshot();
  });
});
