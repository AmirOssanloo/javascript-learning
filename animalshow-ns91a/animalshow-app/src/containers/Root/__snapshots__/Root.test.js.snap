// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root should match with snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.h1>
      Animal Query
    </styled.h1>
    <AnimalQuery
      onChangeToAnimalDisplay={[Function]}
    />
  </styled.div>
</styled.div>
`;
